src/defiat/api.ts

```diff
@@ -447,6 +447,62 @@ export const totalPendingAnyStake = async (
   }
 };
 
+export const totalPendingVirtualAnyStake = async (
+  AnyStake: Contract,
+  pools: StakingPool[],
+  account: string,
+  block: number
+) => {
+  try {
+    var totalPending: BigNumber = new BigNumber(0);
+    for (const pool of pools) {
+      const result = await pendingVirtualAnyStake(AnyStake, pool.pid, account, block);
+      totalPending = totalPending.plus(result);
+    }
+    return new BigNumber(totalPending);
+  } catch (e) {
+    return new BigNumber("0");
+  }
+};
+
+export const pendingVirtualAnyStake = async (
+  AnyStake: Contract,
+  pid: number,
+  account: string,
+  block: number
+) => {
+  try {
+    if (block === 0)
+      return new BigNumber("0");
```

Comment:
nit: I like to put braces around everything for readability (visual separation). I see this a lot in the wild and over time it can become really messy/confusing.